addons/mail/static/src/model/model_core.js

```diff
@@ -341,16 +341,32 @@ export function patchRecordMethods(modelName, recordMethods) {
 }
 
 /**
- * @param {Object} definition The JSON definition of the model to register.
- * @param {Object} [definition.fields]
- * @param {string[]} definition.identifyingFields
- * @param {Object} [definition.lifecycleHooks]
- * @param {Object} [definition.modelGetters] Deprecated; use fields instead.
- * @param {Object} [definition.modelMethods]
- * @param {string} definition.name
- * @param {OnChanges[]} [definition.onChanges]
- * @param {Object} [definition.recordGetters] Deprecated; use fields instead.
- * @param {Object} [definition.recordMethods]
+ * @typedef {Object} LifecycleHooks
+ * @property {Function} [_created] Hook triggered when the model record is created.
+ * @property {Function} [_willCreate] Hook triggered when a model record is going to be created.
+ * @property {Function} [_willDelete] Hook triggered when a model record is about to be deleted.
+ */
+
+/**
+ * @typedef {Object} ModelDefinition
+ * @property {Object} [fields] Fields of the model
+ * @property {string[]} identifyingFields Fields used to uniquely identify a record of this model.
```

Comment:
incomplete type, I'm not sure the syntax but probably Array[string|string[]]

Reply:
I guess `{Array<string|string[]>}` should be better